### PR TITLE
[Python O11y] Build and distrib O11y package

### PR DESCRIPTION
**Pull request by XuanWang-Amos**
Actually build O11y artifacts.

### Testing
* Manually installed the `.whl` and verified it's working locally (For Python 3.8 + Linux).

### CODE REVIEW

File: src/python/grpcio_observability/README.rst [hunk 1]
```diff
@@ -1,2 +1,91 @@
...
+  opentelemetry-sdk==1.21.0
+  opentelemetry-api==1.21.0
+
+
+Usage
+-----
+
+You can find example usage in `Python example folder <https://github.com/grpc/grpc/tree/master/examples/python/observability>`_.
+
+We also provides couple of environment variables to help you optimize gRPC python observability for your particular use.
```
Comment (gnossen) [inline src/python/grpcio_observability/README.rst:74]: provides -> provide

couple of -> several

File: src/python/grpcio_observability/README.rst [hunk 2]
```diff
@@ -1,2 +1,91 @@
-### TODO(xuanwn)
```
Comment (gnossen) [inline src/python/grpcio_observability/README.rst:1]: I think it'd be a good idea to also publish API documentation [like this](https://github.com/grpc/grpc/blob/master/doc/python/sphinx/grpc_health_checking.rst).

File: tools/run_tests/artifacts/build_artifact_python.sh [hunk 3]
```diff
@@ -280,4 +281,14 @@ then
   ${SETARCH_CMD} "${PYTHON}" src/python/grpcio_admin/setup.py \
       sdist bdist_wheel
   cp -r src/python/grpcio_admin/dist/* "$ARTIFACT_DIR"
+
+  # Build grpcio_observability source distribution
+  # Skips MacOS since grpcio_observability does not support MacOS.
+  if [ "$GRPC_UNIVERSAL2_REPAIR" == "" ]; then
```
Comment (gnossen) [inline tools/run_tests/artifacts/build_artifact_python.sh:287]: Having an API test [like this](https://github.com/grpc/grpc/blob/master/src/python/grpcio_tests/tests/unit/_api_test.py#L26) should be a blocker for publishing artifacts. We want to make sure there are no backward-incompatible changes accidentally made.

Comment (gnossen) [inline tools/run_tests/artifacts/build_artifact_python.sh:287]: I see what you're doing here, but the name of this env var is misleading. Can we change the name to `GRPC_BUILD_MAC` or something like that? You'll have to also change [the place where we supply the environment variable](https://github.com/grpc/grpc/blob/1751f1043ed00cb064907bf6943c7fcad72516af/tools/run_tests/artifacts/artifact_targets.py#L145).

File: src/python/grpcio_observability/README.rst [hunk 4]
```diff
@@ -1,2 +1,91 @@
...
+gRPC Python Observability
+===========
+Package for gRPC Python Observability.
+
+More details can be found in `OpenTelemetry Metrics gRFC <https://github.com/grpc/proposal/blob/master/A66-otel-stats.md#opentelemetry-metrics>`_.
+
+How gRPC Python Observability Works
+-------------------------
+
+gRPC Python is a wrapper layer built upon the gRPC Core (written in C/C++). Most of census data
```
Comment (gnossen) [inline src/python/grpcio_observability/README.rst:10]: "Census data"? Outdated?

File: src/python/grpcio_observability/README.rst [hunk 5]
```diff
@@ -1,2 +1,91 @@
...
+is collected at core layer and then exported to Python layer. To optimize performance and reduce
+the overhead of acquiring GIL too frequently, census data is initially cached at the Core layer
```
Comment (gnossen) [inline src/python/grpcio_observability/README.rst:12]: "GIL" -> "the GIL"

File: src/python/grpcio_observability/README.rst [hunk 6]
```diff
@@ -1,2 +1,91 @@
...
+
+
+Supported Python Versions
+-------------------------
+Python >= 3.7
+
+Installation
+------------
+
+Currently gRPC Python Observability is :code:`only available for Linux`.
```
Comment (gnossen) [inline src/python/grpcio_observability/README.rst:26]: Why `:code:` formatting for this? It's not code. [Do you want bold](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/inline.html)?

File: tools/run_tests/artifacts/build_artifact_python.sh (hunk 3 above)
Comment (gnossen) [reply tools/run_tests/artifacts/build_artifact_python.sh]: I mean that we need to have a test that will fail if we remove a symbol from the API. That test will likely not live in the same file.

**Comment by XuanWang-Amos** [comment]
@gnossen PTALA.

**Review by gnossen — APPROVED** [review]

**Comment by gnossen** [comment]
This is a big step! Happy to see this coming along.

**Pull request closed without merging**